include/zenoh_commons.h

```diff
@@ -1027,6 +1041,54 @@ int8_t z_attachment_iterate(struct z_attachment_t this_,
  * Returns the gravestone value for `z_attachment_t`.
  */
 ZENOHC_API struct z_attachment_t z_attachment_null(void);
+/**
+ * Returns `true` if the buffer is in a valid state.
+ */
+ZENOHC_API bool z_buffer_check(const struct z_owned_buffer_t *buffer);
+/**
+ * Increments the buffer's reference count, returning an owned version of the buffer.
+ */
+ZENOHC_API struct z_owned_buffer_t z_buffer_clone(struct z_buffer_t buffer);
+/**
+ * Returns an owned version of this buffer whose data is guaranteed to be contiguous in memory.
+ *
+ * This is achieved by increasing the reference count if the buffer is already contiguous, and by copying its data in a new contiguous buffer if it wasn't.
+ */
+ZENOHC_API
+struct z_owned_buffer_t z_buffer_contiguous(struct z_buffer_t buffer);
```

Comment:
This is similar situation to mutable API discussed previously. Implicit copy API is bad for the following reasons:
- masks the potential operation overhead (for user code reviewers)
- isolates user code from having knowledge on the side effects
- forces user code to have only one behavior (what if user wants to drop non-contiguous buffers instead of copying them? or user wants to process non-contiguous buffers in separate code branch? or user doesn't need to create `z_owned_buffer_t` wrapper around contiguous data?)

Instead, I just suggest adding `bool is_contiguous(struct z_buffer_t);` convenience method and give user all the freedom.

Reply:
As discussed at the meeting today, I support the idea of removing z_buffer_contiguous both here and in rust.
For simple cases (number, string, etc., which IMHO will cover most of the needs of ordinary users), simple operations for obtaining them should be provided, for complex ones (working directly with buffers, which will most likely be used by experienced users) we should not hide the internal structure to avoid side effects.